shared/src/utils/validators/get-map-validator.ts

```diff
@@ -0,0 +1,43 @@
+export interface ValidationFunctions {
+    keyValidator: (key: string) => boolean;
+    /**
+     * This validator must return `false` when {@link value} is not a valid object.
+     */
+    valueValidator: (value: unknown) => boolean;
+    consistencyValidator?: (key: string, value: unknown) => boolean;
+}
+
+// TODO: The formatting of the list is ugly as something removes the trailing whitespaces that would make it create newlines.
+/**
+ * The provided flow is as follows:
+ *
+ * 1. It's checked whether `valueToBeValidated` is of type `object`
+ * 2. It's checked whether `valueToBeValidated` is not `null`
+ * 3. For every key-value pair of the object, the following are verified:
+ *
+ *      a. It's checked whether the key fulfills {@link keyValidator}
+ *
+ *      b. The value gets converted to a (suspected, see {@link valueTransformer} and {@link valueValidator}) object of the target type using {@link valueTransformer}
+ *
+ *      c. It's checked whether the transformed value is valid using {@link valueValidator}
+ *
+ *      d. When {@link consistencyValidator} is provided, it's applied to the key and value
+ *
```

Comment:
```suggestion
/**
 * The provided flow is as follows:
 *
 * 1. It's checked whether `valueToBeValidated` is of type `object`
 * 2. It's checked whether `valueToBeValidated` is not `null`
 * 3. For every key-value pair of the object, the following are verified:
 *    1. It's checked whether the key fulfills {@link keyValidator}
 *    2. The value gets converted to a (suspected, see {@link valueTransformer} and {@link valueValidator}) object of the target type using {@link valueTransformer}
 *    3. It's checked whether the transformed value is valid using {@link valueValidator}
 *    4. When {@link consistencyValidator} is provided, it's applied to the key and value
 *
```

See https://github.com/markdown-it/markdown-it/issues/215